packages/build/src/steps/run_core_step.js

```diff
@@ -0,0 +1,209 @@
+ * @param  {string} [flags.siteId] - Netlify Site ID
+ * @param  {string} [flags.deployId] - Netlify Deploy ID
+ * @param  {string} [flags.context] - Build context
+ * @param  {string} [flags.branch] - Repository branch
+ * @param  {boolean} [flags.dry=false] - Run in dry mode, i.e. printing steps without executing them
+ * @param  {string} [flags.nodePath] - Path to the Node.js binary to use in the build command and plugins
+ * @param  {boolean} [flags.buffer=false] - Buffer output instead of printing it
+ *
+ * @returns {object} buildResult
+ * @returns {boolean} buildResult.success - Whether build succeeded or failed
+ * @returns {number} buildResult.severityCode - Build success/failure status among:
+ * 0 (success), 1 (build cancelled), 2 (user error), 3 (plugin error), 4 (system error). Can be used as exit code.
+ * @returns {string[]} buildResult.logs - When using the `buffer` option, all log messages
+ */
+export default async function runCoreStep(buildSteps, flags = {}) {
+  const { errorMonitor, mode, logs, debug, ...flagsA } = startBuild(flags)
+
+  const errorParams = { errorMonitor, mode, logs, debug }
+
+  try {
+    const { netlifyConfig: netlifyConfigA, configMutations } = await executeBuildStep({
+      ...flagsA,
+      errorMonitor,
+      mode,
+      logs,
+      debug,
+      errorParams,
+      buildSteps,
+    })
+
+    const { success, severityCode } = getSeverity('success')
+
+    return { success, severityCode, netlifyConfig: netlifyConfigA, configMutations, logs }
+  } catch (error) {
+    const { severity } = await handleBuildError(error, errorParams)
+
+    const { success, severityCode } = getSeverity(severity)
+
+    return { success, severityCode, logs }
+  }
+}
+
+const startBuild = function (flags) {
+  const logs = getBufferLogs(flags)
+
+  logBuildStart(logs)
+
+  const { bugsnagKey, ...flagsA } = normalizeFlags(flags, logs)
+
+  const errorMonitor = startErrorMonitor({ flags: flagsA, logs, bugsnagKey })
+
+  return { ...flagsA, errorMonitor, logs }
+}
+
+const getBuildSteps = function (buildSteps) {
+  const allSteps = []
+
+  if (buildSteps.includes('edgeFunctionsBundling')) {
+    // eslint-disable-next-line fp/no-mutating-methods
+    allSteps.push(bundleEdgeFunctions)
+  }
+
+  return allSteps
+}
+
+const executeBuildStep = async function ({
+  config,
+  defaultConfig,
+  cachedConfig,
+  debug,
+  nodePath,
+  functionsDistDir,
+  edgeFunctionsDistDir,
+  errorMonitor,
+  mode,
+  logs,
+  errorParams,
+  featureFlags,
+  buildSteps,
+  repositoryRoot,
+}) {
+  const configOpts = getConfigOpts({
+    config,
+    defaultConfig,
+    featureFlags,
+    mode,
+    repositoryRoot,
+  })
+
+  const {
+    netlifyConfig,
+    buildDir,
+    siteInfo,
+    childEnv,
+    userNodeVersion,
+    repositoryRoot: repositoryRootA,
+  } = await loadConfig({
+    configOpts,
+    cachedConfig,
+    debug,
+    logs,
+    nodePath,
+    timers: [],
+  })
+
```

Comment:
nit: Removing empty line between declarations inside method, to reduce verbosity and keep consistency with the rest of the codebase.

```suggestion
```